app/src/main/java/ch/epfl/sdp/blindly/main_screen/profile/ProfilePageFragment.kt

```diff
@@ -0,0 +1,119 @@
+        private const val TAG = "ProfilePage"
+        const val BOUNCE_DURATION: Long = 100
+        private const val ARG_COUNT = "profileArgs"
+        private var counter: Int? = null;
+
+        fun newInstance(counter: Int): ProfilePageFragment {
+            val fragment = ProfilePageFragment();
+            val args = Bundle();
+            args.putInt(ARG_COUNT, counter);
+            fragment.arguments = args;
+            return fragment
+        }
+    }
+
+    /**
+     * Set up the viewModel
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (arguments != null) {
+            counter = requireArguments().getInt(ARG_COUNT)
+        }
+
+        val bundle = Bundle()
+        bundle.putString("uid", userHelper.getUserId())
+
+        val viewModelFactory = assistedFactory.create(this, bundle)
+
+        viewModel = ViewModelProvider(this, viewModelFactory)[ProfilePageViewModel::class.java]
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+
+        return inflater.inflate(R.layout.fragment_profile_page, container, false)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val userInfoText = view.findViewById<TextView>(R.id.user_info_text)
+        val userDescriptionText = view.findViewById<TextView>(R.id.user_description_text)
+
+        val context = this@ProfilePageFragment.context
+
+        val editButton = view.findViewById<Button>(R.id.edit_info_profile_button)
+        setOnClickListener(editButton, Intent(context, EditProfile::class.java))
+
+        val recordAudioButton = view.findViewById<Button>(R.id.record_audio_profile_button)
+        setOnClickListener(recordAudioButton, Intent(context, RecordingActivity::class.java))
+
+        val settingsButton = view.findViewById<Button>(R.id.settings_profile_button)
+        setOnClickListener(settingsButton, Intent(context, Settings::class.java))
+
+        viewModel.user.observe(viewLifecycleOwner) {
+            userInfoText.text = "${it.username}, ${User.getUserAge(it)}"
+            if(it.description == "") {
+                userDescriptionText.text = "Add description"
+            } else {
+                userDescriptionText.text = it.description
+            }
+        }
+    }
+
+
+    /**
+     * An onClickListener that start an Activity after the button has stopped bouncing
+     * @param button: the button associated with the click
+     * @param intent: the intent containing the next Activity to start
+     */
+    private fun setOnClickListener(button: Button, intent: Intent) {
+        val bounce = AnimationUtils.loadAnimation(context, R.anim.bouncy_button)
+        button.setOnClickListener {
+            button.startAnimation(bounce)
```

Comment:
Since we use the bouncing a lot, it would be nice to integrate it in the animations package instead of writing this code everytime we want to use it. It can be done later though, when we plan on refactoring.

Reply:
Yes great idea!